packages/keyring-api/src/utils/pagination.ts

```diff
@@ -0,0 +1,17 @@
+import type { Infer } from '@metamask/superstruct';
+import { number, string } from '@metamask/superstruct';
+import { exactOptional, object } from '@metamask/utils';
+
+export const PaginationStruct = object({
+  /**
+   * Maximum number of items to return.
+   */
+  limit: number(),
+
+  /**
+   * Next cursor to iterate over the results.
+   */
+  next: exactOptional(string()),
```

Comment:
I wonder if we should `optional(nullable(string()))` here...

I was thinking of this example
```ts
const keyring = ...;

const pagination = {
  limit: 10,
  next: null, // No cursor yet
};
do {
  const page = await keyring.listAccountTransactions(accountId, pagination);
  // Do something with `page.transactions`...
  pagination.next = page.next;
} while (page.next);
```

I think we cannot write this code here, because of the `exactOptional`, right?

Reply:
This type was supposed to be used in the request, not the response:

```ts
listAccountTransactions(id, {limit: 10});
// or
listAccountTransactions(id, {limit: 10, next: '<cursor_token>'});
```

But now that you mentioned this case, I think we can do the change you mentioned and repurpose it to be used both ways 🤔 WDYT?

Reply:
Well... I really liked the `exactOptional` initially yes, but I'm worried about it being hard to use, so I believe we could just allow `null` in this case yes.